test/spec/protractor_sync_test.ts

```diff
@@ -88,6 +94,16 @@ describe('Protractor extensions', () => {
       testMultilineInput = elementSync.findVisible('textarea');
     }));
 
+    it('causes large heap snapshot on error', createTest(() => {
+      heapdump.writeSnapshot('/Users/mwhitsitt/heaps/beforeFailure.heapsnapshot');
+      polledExpect(() => { return false; }).toBe(true);
+    }));
```

Comment:
this test case generates a 'before' heap snapshot that is about 17MB in file size but the 'after' heap snapshot is 101MB file size. Of course the actual heap sizes are slight different.

I haven't yet been able to open my larger heap snapshot to see what is in it.